gm_dotnet_native/dotnethelper-src/dotnethelper.cpp

```diff
@@ -36,40 +41,37 @@ typedef cleanup_function_fn(*managed_main_fn)(
         /* Out Param */ managed_delegate_executor_fn* managed_delegate_executor_ptr
         );
 
+using tstring = std::basic_string<char_t>;
+
 std::ofstream error_log_file;
 
 managed_delegate_executor_fn managed_delegate_executor = nullptr;
 
 managed_main_fn managed_main = nullptr;
 
-std::string hosfxr_path = "garrysmod/lua/bin/dotnet/host/fxr/" + std::string(NET_CORE_VERSION);
+const std::filesystem::path bin_folder = _T("garrysmod/lua/bin");
```

Comment:
`bin_folder` var name here can be little bit confusing: Garry's Mod has two bin folders, one for Source Engine binaries and one for Lua native plugins, which we are taking about. I would rather rename it to `lua_bin_folder`.

Reply:
I made the changes as you suggested. 

Other than that I would probably propose moving library loading code into a shared header.